drivers/spi/spi_ll_stm32.c

```diff
@@ -90,9 +99,26 @@ static void spi_stm32_shift_m(SPI_TypeDef *spi, struct spi_stm32_data *data)
 	u16_t rx_frame;
 
 	tx_frame = spi_stm32_next_tx(data);
+
+#ifdef CONFIG_SOC_SERIES_STM32MP1X
+	/* STM32MP1 has some register name modifications :
+	 * LL_SPI_IsActiveFlag_TXP <-> LL_SPI_IsActiveFlag_TXE
```

Comment:
In that case, might be good to add a level of abstraction here instead of ifdefing the whole code, cd what was done here:
https://github.com/zephyrproject-rtos/zephyr/tree/master/drivers/ipm/ipm_stm32_ipcc.c

Reply:
That's a good idea, I'll do that ! 
The equivalence is not straight forward, but I think the behavior should be the same  :

**Other MCUs : TXE**
0: Tx buffer not empty
1: Tx buffer empty
**MP1 : TXP**
0: there is not enough space to locate next data packet at TxFIFO
1: TxFIFO has enough free location to host 1 data packet
_________________________________________

**Other MCUs : RXNE**
0: Rx buffer empty
1: Rx buffer not empty
**MP1 : RXP**
0: RxFIFO is empty or a not complete data packet is received
1: RxFIFO contains at least 1 data packet